Conceptpower+Spring/src/main/java/edu/asu/conceptpower/users/impl/UsersManager.java

```diff
@@ -24,17 +30,41 @@
  * @author Julia Damerow
  * 
  */
+@PropertySource(value = "classpath:/user.properties")
 @Service
 public class UsersManager implements IUserManager {
 
 	@Autowired
 	private UserDatabaseClient client;
-	private Map<String, String> admins;
-
-	@PostConstruct
-	public void init() {
-		admins = new HashMap<String, String>();
-	}
+	private Map<String, Object> admins;
+	
+	
+	@Autowired
+    Environment env;
+	
+	
+    @PostConstruct
+    public void init() throws IOException {
+        admins = new HashMap<String, Object>();
+        for (Iterator it = ((AbstractEnvironment) env).getPropertySources().iterator(); it.hasNext();) {
+            Object source = (Object) it.next();
+            if (source instanceof ResourcePropertySource) {
+                ResourcePropertySource propertySource = (ResourcePropertySource) source;
+                admins.putAll(((ResourcePropertySource) propertySource).getSource());
```

Comment:
not sure if that's the best way of doing it. Why don't you call getPropertyNames() on the propertysource and iterate over the returned array of username here and add them to the admins map right away (loose the loop in line 61).